diff --git a/libavformat/movenc.c b/libavformat/movenc.c.in
index 885e4c4..76dce9e 100644
--- a/libavformat/movenc.c
+++ b/libavformat/movenc.c.in
@@ -8606,7 +8606,6 @@ static const AVCodecTag codec_mp4_tags[] = {
     { AV_CODEC_ID_MJPEG,           MKTAG('m', 'p', '4', 'v') },
     { AV_CODEC_ID_PNG,             MKTAG('m', 'p', '4', 'v') },
     { AV_CODEC_ID_JPEG2000,        MKTAG('m', 'p', '4', 'v') },
-    { AV_CODEC_ID_JPEGXS,          MKTAG('j', 'x', 's', ' ') },
     { AV_CODEC_ID_VC1,             MKTAG('v', 'c', '-', '1') },
     { AV_CODEC_ID_DIRAC,           MKTAG('d', 'r', 'a', 'c') },
     { AV_CODEC_ID_TSCC2,           MKTAG('m', 'p', '4', 'v') },
