diff --git a/libavcodec/codec_id.h b/libavcodec/codec_id.h.in
index 42a2bd5..2f6efe8 100644
--- a/libavcodec/codec_id.h
+++ b/libavcodec/codec_id.h.in
@@ -329,7 +329,6 @@ enum AVCodecID {
     AV_CODEC_ID_DNXUC,
     AV_CODEC_ID_RV60,
     AV_CODEC_ID_JPEGXL_ANIM,
-    AV_CODEC_ID_JPEGXS,
 
     /* various PCM "codecs" */
     AV_CODEC_ID_FIRST_AUDIO = 0x10000,     ///< A dummy id pointing at the start of audio codecs
