diff --git a/libavcodec/codec_desc.c b/libavcodec/codec_desc.c.in
index ad1e094..9fb190e 100644
--- a/libavcodec/codec_desc.c
+++ b/libavcodec/codec_desc.c.in
@@ -1985,15 +1985,6 @@ static const AVCodecDescriptor codec_descriptors[] = {
         .props     = AV_CODEC_PROP_LOSSY | AV_CODEC_PROP_LOSSLESS,
         .mime_types= MT("image/jxl"),
     },
-    {
-        .id        = AV_CODEC_ID_JPEGXS,
-        .type      = AVMEDIA_TYPE_VIDEO,
-        .name      = "jpegxs",
-        .long_name = NULL_IF_CONFIG_SMALL("JPEG XS"),
-        .props     = AV_CODEC_PROP_INTRA_ONLY | AV_CODEC_PROP_LOSSY |
-                     AV_CODEC_PROP_LOSSLESS,
-        .mime_types= MT("image/jxs"),
-    },
 
     /* various PCM "codecs" */
     {
